app.py

```
@@ -40,57 +42,93 @@ def ping():
-
-        if not serial_connection.open_comport(): return "Failed to open comport"
+        with serial_lock():
+            # This check can cause a race condition. It needs to be in the lock.
+            if serial_connection.target is not None:
+                connected = True
+            else:
+                serial_connection.init_comport(name=name, baudrate=921600, timeout=timeout)
 
-        # send connect opcode
-        serial_connection.connect()
+                if not serial_connection.open_comport(): return Response("Failed to open comport", status=400)
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)